tests/integration/test_extensions.py

```diff
@@ -282,6 +282,50 @@ def test_valloc_at_thread_exit(tmpdir, monkeypatch):
     assert len(vallocs) == 1
 
 
+def test_valloc_at_thread_exit_in_subprocess(tmpdir, monkeypatch):
```

Comment:
Let's decorate this with the valgrind marker

Reply:
It wouldn't help, as the meat of it is run in a subprocess, and we don't run valgrind with `--trace-children=yes`.

It's run in a subprocess so that we can control the order in which the different TLS variables get created - the non-subprocess one will always have done an `import memray` first, so the subprocess one is for checking the case where an extension with a TLS destructor that allocates memory creates its TLS variable before Memray has created its.

Reply:
That said - maybe we _should_ use `--trace-children=yes` for Valgrind?

Reply:
Maybe in a different PR.